public/scripts/reasoning.js

```diff
@@ -1212,6 +1324,31 @@ function registerReasoningAppEvents() {
     }
 }
 
+/**
+ * Loads reasoning templates from the settings data.
+ * @param {object} data Settings data
+ * @param {ReasoningTemplate[]} data.reasoning Reasoning templates
+ * @returns {Promise<void>}
+ */
+export async function loadReasoningTemplates(data) {
+    if (data.reasoning !== undefined) {
+        reasoning_templates.splice(0, reasoning_templates.length, ...data.reasoning);
+    }
+
+    for (const template of reasoning_templates) {
+        $('<option>').val(template.name).text(template.name).appendTo(UI.$select);
+    }
+
+    if (!power_user.reasoning.name) {
+        power_user.reasoning.name = reasoning_templates[0]?.name ?? '';
+    }
+
+    UI.$select.val(power_user.reasoning.name);
```

Comment:
We have an a slight issue here.
Generally it is correct to only load the template name name, and not trigger change to refresh the other values from the template.

Though for the first load/migration here I am not sure if it makes sense.
You are automatically choosing to apply a template name (namely the one we add here as the default), but the user might have chosen something totally different for his reasoning template values.

I myself was pretty confused, wondering why the template looked like my customized one.

Not sure I have a good solution. Keeping the dropdown empty at first?
Automatically creating a template for the user on the first-time when no name was set but the users' fields are different than the default template?
I think the latter could work.